msal/src/main/java/com/microsoft/identity/client/GlobalSettings.java

```diff
@@ -0,0 +1,173 @@
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client;
+
+import static com.microsoft.identity.client.GlobalSettingsConfigurationFactory.initializeGlobalConfiguration;
+import static com.microsoft.identity.client.exception.MsalClientException.SAPCA_USE_WITH_MULTI_POLICY_B2C;
+import static com.microsoft.identity.client.internal.MsalUtils.validateNonNullArgument;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+
+import com.microsoft.identity.client.configuration.AccountMode;
+import com.microsoft.identity.client.exception.MsalClientException;
+import com.microsoft.identity.client.exception.MsalException;
+import com.microsoft.identity.common.java.authorities.AzureActiveDirectoryB2CAuthority;
+import com.microsoft.identity.common.logging.Logger;
+
+import java.io.File;
+
+/**
+ * Class used to initialize global configurations for the library.
+ */
+public class GlobalSettings {
+    private static final String TAG = GlobalSettings.class.getSimpleName();
+    public static final String NO_GLOBAL_SETTINGS_WARNING = "Global settings have not been initialized before the creation of this PCA Configuration.";
+    public static final String GLOBAL_INIT_AFTER_PCA = "pca_created_before_global";
+    public static final String GLOBAL_INIT_AFTER_PCA_ERROR_MESSAGE = "Global initialization was attempted after a PublicClientApplicationConfiguration instance was already created. Please initialize global settings before any PublicClientApplicationConfiguration instance is created.";
+
+    private static GlobalSettingsConfiguration mGlobalSettingsConfiguration;
+    private static boolean mPCAConfigCreated = false;
+    private static boolean mGlobalSettingsInitialized = false;
+    private static Object mGlobalSettingsLock = new Object();
+
+    /**
+     * Load the global configuration file using the context, resource id of the configuration file, and a listener.
+     *
+     * @param context Context of the app.
+     * @param configFileResourceId Resource Id for the configuration file.
+     * @param listener Handles success and error messages.
+     */
+    public static void loadGlobalConfigurationFile(@NonNull final Context context,
+                                                   final int configFileResourceId,
+                                                   @NonNull final GlobalSettingsListener listener) {
+        validateNonNullArgument(context, PublicClientApplication.NONNULL_CONSTANTS.CONTEXT);
+        validateNonNullArgument(listener, PublicClientApplication.NONNULL_CONSTANTS.LISTENER);
+        runOnBackground(new Runnable() {
```

Comment:
we probably want to lock when the method starts, and unlock when the background task finish.